### PR TITLE
fixes for select-menu with `outline` appearance

### PR DESCRIPTION
**Pull request by curryy**

### CODE REVIEW

File: packages/menu/src/styles.tsx [hunk 1]
```diff
@@ -43,28 +53,52 @@ export const Item = styled.div<ItemProps>`
...
-        ? `0 -1px 0 0 ${_.theme.colors.border.default}`
-        : undefined
-    })}
-  }
+  ${_ =>
+    _.separated && {
+      '&:hover': {
+        [`${Item}:not([data-isselectable='false'])`]: {
+          [Text]: {
+            transition: 'color 0.25s ease',
```
Comment (hzub) [inline packages/menu/src/styles.tsx]: Please note that transition works only when mouse is entering the field, not the other way - in order to fix it, move the attribute out of `&:hover` scope

File: packages/menu/src/styles.tsx [hunk 2]
```diff
@@ -43,28 +53,52 @@ export const Item = styled.div<ItemProps>`
...
+            color: _.invalid
+              ? _.theme.colors.text.danger
+              : _.theme.colors.text.muted,
```
Comment (hzub) [inline packages/menu/src/styles.tsx]: "unhovered" items looks the same as disabled ones - is it correct according to UX?

File: packages/select-menu/src/types.ts [hunk 3]
```diff
@@ -89,6 +89,17 @@ export interface SelectMenuProps<T extends OptionBase> {
    */
   width?: number
   // minWidth?: number
+
+  /**
+   * Are items selectable
+   */
+  isSelectable?: boolean
```
Comment (hzub) [inline packages/select-menu/src/types.ts]: Summing up the logic, `closeOnSelect` implies that `isSelectable` as `false` - if we can't select something (in a lasting way) then we should hide checkboxes anyway. In other words: we can merge those two props into one, just name it properly.

**Review by hzub — CHANGES_REQUESTED** [review]

File: packages/menu/src/styles.tsx (hunk 2 above)
Comment (curryy) [reply packages/menu/src/styles.tsx]: Hmmm. If select is disabled - the menu won't show up, so menu items haven't disabled state.

File: packages/menu/src/index.tsx [hunk 4]
```diff
@@ -322,6 +323,12 @@ export interface MenuOptionsGroupProps {
    * Separate items with a border
    */
   separated?: boolean
+
+  /**
+   * Items ara selectable
```
Comment (hzub) [inline packages/menu/src/index.tsx]: typo

**Review by hzub — APPROVED** [review]

**Pull request merged**